### PR TITLE
[RLlib] Move connectors creation code to single point in RolloutWorker

### PR DESCRIPTION
**Pull request by ArturNiederfahrenhorst**
Signed-off-by: Artur Niederfahrenhorst <artur@anyscale.com>

## Why are these changes needed?

My first connectors/filters PR had two spots to have connectors, where there should be only one ideally.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by gjoliver — APPROVED** [review]
yeah, that's right!! this is awesome!!
thanks.

**Comment by gjoliver** [comment]
can you take a look at the failed tests? test_rollout_worker looks suspicious.

**Comment by ArturNiederfahrenhorst** [comment]
test_rollout_worker was marked as flakey and according to https://flakey-tests.ray.io/ has already been flakey. I can't reproduce an error on my side.

**Comment by c21** [comment]
Can it be merged? What's the remaing work here?

**Comment by ArturNiederfahrenhorst** [comment]
No, thank you. Not a pressing change.

**Comment by c21** [comment]
Cool, removed as a blocker for branch cut.

File: rllib/evaluation/rollout_worker.py [hunk 1]
```diff
@@ -1905,9 +1868,24 @@ def _build_policy_map(
                     merged_conf,
                 )
 
-            if connectors_enabled and name in self.policy_map:
-                create_connectors_for_policy(self.policy_map[name], policy_config)
+            new_policy = self.policy_map[name]
```
Comment (ArturNiederfahrenhorst) [inline rllib/evaluation/rollout_worker.py:1871]: _build_policy_map is called in add_policy and in init.

**Pull request merged**